### PR TITLE
Intake mech

### PR DESCRIPTION
**Pull request by adnx26**

### CODE REVIEW

File: src/main/java/frc/robot/systems/monkeFSM.java [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+	private void handleStartIdleState(TeleopInput input) {
+		armMotor.set(0);
+		armSolenoid.set(Value.kReverse);
+	}
+	/**
+	 * Handle behavior in first retract state.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	private void handleFirstRetractState(TeleopInput input) {
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/monkeFSM.java:288]: You should also set the solenoid in every state handler as well, to make it clear what the intended position is.

File: .gitignore [hunk 2]
```diff
@@ -162,3 +162,38 @@ imgui.ini
 
 
 # End of https://www.gitignore.io/api/c++,java,linux,macos,gradle,windows,visualstudiocode
+local.properties
+.idea/workspace.xml
```
Comment (AG6GR) [inline .gitignore:166]: Just add one entry for `.idea` to ignore the entire folder.

**Review by AG6GR — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/systems/monkeFSM.java [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private FSMState nextState(TeleopInput input) {
```
[the remark this answers is not in the document]
Comment (andrewqin29) [reply src/main/java/frc/robot/systems/monkeFSM.java:152]: diagram committed to repo

File: src/main/java/frc/robot/systems/monkeFSM.java [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+	private static int CYCLE_COUNT = 0;
+
+	/* ======================== Private variables ======================== */
+	private FSMState currentState;
+
+	// Hardware devices should be owned by one and only one system. They must
+	// be private to their owner system and may not be used elsewhere.
+	private CANSparkMax armMotor;
+	private DoubleSolenoid armSolenoid;
+	private DigitalInput armLimitSwitchFirst;
```
[the remark this answers is not in the document]
Comment (andrewqin29) [reply src/main/java/frc/robot/systems/monkeFSM.java]: we thought we would need one for fully retracted, the little extension, as well as the extension to swing onto the next bar (handleArmPistonExtendState) for a combined total of 3. we haven't had a thorough discussion with hardware team regarding limit switches yet.

[the remark this answers is not in the document]
Comment (andrewqin29) [reply src/main/java/frc/robot/systems/monkeFSM.java]: also, wasn't really sure how to declare the limit switches from the documentation provided. I committed and pushed my best guess onto the repo so could you check that. the only cansparkmax we declared was armMotor so I just did armMotor.getForwardLimitSwitch() for all three limitswitches. but I'm confused on why the armMotor would be responsible for connecting to the limitswitches in the first place.

**Review by AG6GR — APPROVED** [review]

**Pull request merged**